### PR TITLE
feat(erc20): Bank Module backed Polaris ERC20 tokens

### PR DESCRIPTION
**Pull request by calbera**
Requires changes to:
- PolarisERC20 contract implementation: no longer solmate, no minting/burning exposed via EVM, using bank/authz module precompiles 
- Authz Module Precompile (special cases)
- Modified ERC20 module precompile to use bank module for PolarisERC20 operations

### CODE REVIEW

File: contracts/src/cosmos/PolarisERC20.sol [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    /**
+     * @dev transfer is a public method for transferring tokens to a given address.
+     * @param to the address to transfer tokens to.
+     * @param amount the amount of tokens to transfer.
+     * @return bool true if the transfer was successful.
+     */
+    function transfer(address to, uint256 amount) public virtual returns (bool) {
+        IBankModule.Coin[] memory coins = amountToCoins(amount);
+        bank().send(msg.sender, to, coins);
```
Comment (itsdevbear) [inline contracts/src/cosmos/PolarisERC20.sol:125]:
```suggestion
	require(bank().send(msg.sender, to, coins), "failed to send bank tokens");
```

File: contracts/src/cosmos/PolarisERC20.sol [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+     * @param amount the amount of tokens to transfer.
+     * @return bool true if the transfer was successful.
+     */
+    function transferFrom(address from, address to, uint256 amount) public virtual returns (bool) {
+        // TODO: Use allowance once authz precompile is available.
+        uint256 allowed = allowance[from][msg.sender]; // Saves gas for limited approvals.
+
+        if (allowed != type(uint256).max) allowance[from][msg.sender] = allowed - amount;
+
+        bank().send(from, to, amountToCoins(amount));
```
Comment (itsdevbear) [inline contracts/src/cosmos/PolarisERC20.sol:144]:
```suggestion
        require(bank().send(msg.sender, to, coins), "failed to send bank tokens");
```

File: contracts/src/cosmos/PolarisERC20.sol [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+                ),
+                v,
+                r,
+                s
+            );
+
+            require(recoveredAddress != address(0) && recoveredAddress == owner, "PolarisERC20: INVALID_SIGNER");
+
+            require(
+                authz().setSendAllowance(recoveredAddress, spender, amountToAuthCoins(value), 0),
```
Comment (itsdevbear) [inline contracts/src/cosmos/PolarisERC20.sol:197]: @zmanian authz via EIP-2612

**Comment by calbera** [comment]
For any IBC bridged SDK coins, we need to ensure setting denom metadata for the incoming coin

**Comment by itsdevbear** [comment]
Good to merge, I'd say add a fuckton more integration tests tho

**Pull request merged**